iroha/src/sumeragi.rs

```diff
@@ -82,6 +82,19 @@ impl Sumeragi {
         })
     }
 
+    /// Updates network topology by taking the actual list of peers from `WorldStateView`.
+    /// Updates it only if the new peers were added, otherwise leaves the order unchanged.
+    pub async fn update_network_topology(&mut self) {
+        let wsv_peers = self.world_state_view.read().await.peer.peers.clone();
+        let sumeragi_peers: HashSet<_> =
+            self.network_topology.sorted_peers.iter().cloned().collect();
+        if wsv_peers != sumeragi_peers {
+            self.network_topology.sorted_peers = wsv_peers.iter().cloned().collect();
+            self.network_topology
+                .sort_peers(Some(self.latest_block_hash));
+        }
+    }
```

Comment:
I would suggest to move main logic of this function inside network topology, so this function will look something like this:
```
self.network_topology.update(self.world_state_view.read().await.peer.peers.clone());
```
This way we will move topology logic inside NetworkTopology.

Reply:
Mostly fixed this comment, though not entirely as for it to be exactly like that we need to move `latest_block_hash` into `NetworkTopology`. I left a TODO to fix it later.